### PR TITLE
Implement `package` on class diagram

### LINKED ISSUE #1052: Add `package` to class diagram
**Is your feature request related to a problem? Please describe.**

Hello! *Иon-native English speaker warning*

I'm relative frequently drawing class diagrams. For now I'm using MagicDraw but I would like to try text-to-scheme solution.

I like your syntax for class diagrams and I would be very happy to use it for my work but I frequently using `packages` in my schemes. Like `class A` goes to `package X` and associated with `class B` that laying in `package Y`. 

**Describe the solution you'd like**

I would like to have a `package` extension for current syntax like:

```
package Sales {
  class Product {
    +int id
  }
  Product o-- Employees.Manager

package Employees {
  class Manager {
    +int id
  }
}
```

**Describe alternatives you've considered**
Do not see any

**Additional context**
None

### PR DESCRIPTION
**Pull request by ksilverwall**
## :bookmark_tabs: Summary
=

Sample Implementation of `package` of class diagram
Resolves #1052

NOTE: the parser logic only, need to implement rendering logic.

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :notebook: have added documentation (if appropriate)
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/diagrams/class/classDiagram.spec.ts
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classDiagram.spec.ts:601]:
```suggestion
      const str = `classDiagram
namespace Namespace1 { class Class1 }
namespace Namespace2 { class Class1
}
namespace Namespace3 {
class Class1 {
int : test
string : foo
test()
foo()
}
}
namespace Namespace4 {
class Class1 {
int : test
string : foo
test()
foo()
}
class Class2 {
int : test
string : foo
test()
foo()
}
}`;
```

File: packages/mermaid/src/diagrams/class/classDiagram.spec.ts
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classDiagram.spec.ts:1021]: Please use ` the same way as the test above.

File: packages/mermaid/src/diagrams/class/classTypes.ts
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classTypes.ts:61]:
```suggestion
export interface NamespaceNode {
  id: string;
  domId: string;
  classes: ClassMap;
  children: NamespaceMap;
};
```

File: packages/mermaid/src/diagrams/class/classDb.ts
Comment (sidharthv96) [inline packages/mermaid/src/diagrams/class/classDb.ts:112]:
```suggestion
  namespaces = {};
  namespaceCounter = 0;
```

**Review by sidharthv96 — CHANGES_REQUESTED** [review]
@ksilverwall nice work so far, just some minor suggestions. 
@knsv please review the Jison changes.

**Comment by ksilverwall** [comment]
@sidharthv96 thank you for your review! I applied all of your suggestions.

**Comment by ksilverwall** [comment]
@sidharthv96 Thank you for your review! I applied all suggestions.

**Comment by sidharthv96** [comment]
@ksilverwall can you avoid force pushing, it's better to have individual smaller commits.

**Comment by ksilverwall** [comment]
I added the rendering logic

**Comment by jgreywolf** [comment]
@ksilverwall Resolve the merge conflicts and let get this out there!

**Comment by ksilverwall** [comment]
@jgreywolf 
I resolved. 🙌

**Comment by ksilverwall** [comment]
@jgreywolf 
I do not have write-access permission to this repository then I cannot merge.
can you merge this?

**Comment by sidharthv96** [comment]
This has significant changes in jison files. Let's wait for a review from @knsv.

**Review by aloisklink — COMMENTED** [review]
From what I've seen, the changes look great, but I'm not really an expert on SVG/jison code, so I'll let Knut look at that.

**Minor nitpick, the colors of the namespace block seem a bit off** (it's very hard to read the namespace title **BaseShapes**):

image

Unfortunately, I don't really know much about CSS/SVGs, so I can't tell you how to change the colors (I'm guessing you need to add something here: https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/diagrams/class/styles.js)

The namespace background color should have a sufficient contrast against the namespace text, so that even people with visual impairments (or people viewing Mermaid diagrams outside/with low-quality screens) can read it.

You can use something like https://webaim.org/resources/contrastchecker/ to check whether the colors are okay.

**Comment by ksilverwall** [comment]
@aloisklink 
Thank you for your review!
I changed namespace rect style with no fill and black border line.

namespace-rect

**Review by jgreywolf — APPROVED** [review]
It is looking good to me. I would like one more review from @knsv on the jison grammar.  Ill bug him about it :)

**Review by aloisklink — APPROVED** [review]
> @aloisklink
> Thank you for your review!
> I changed namespace rect style with no fill and black border line.

**Awesome!** Thanks for making those changes! It even looks good in dark mode: 
image

File: packages/mermaid/src/diagrams/class/classRenderer-v2.ts
Comment (aloisklink) [reply packages/mermaid/src/diagrams/class/classRenderer-v2.ts:89]: Fixed by https://github.com/mermaid-js/mermaid/pull/4206/commits/0d373f3a6ab7c8f858b243bfc76677f70b93ade6, thanks @ksilverwall :)

**Pull request merged**
